### ISSUE TITLE
KIE doesn't work with more accurate text readings

### ISSUE THREAD

**Issue by parulsingh23** [open-mmlab/mmocr#415]
I replaced the text detection & recognition pipeline with my own model (as I wanted to catch some misread words & undetected text). However, with plugging this output into KIE, I'm getting an even worse identification of receipt fields.

Here is the receipt with my custom model running det & rec:
![image](https://user-images.githubusercontent.com/55006754/128650051-f34687e2-5633-43db-9bc9-1608dad5ef01.png)

And when I provide this to KIE, it returns:
```
bakery  is  Store_name_value
bakery  is  Store_name_value
50  is  Prod_price_value
50  is  Subtotal_value
subtotal  is  Subtotal_key
40  is  Tax_value
46  is  Tax_value 
tax  is  Tax_key
total  is  Total_key
96  is  Total_value (from '16.96' or ‘$6.96’)
96  is  Total_value
total  is  Total_key
nttpf  is  Prod_quantity_key
```
However, simply plugging in the original receipt image into `ocr = MMOCR(det='PS_CTW', recog='SAR', kie='SDMGR')`, I get much better results:
```
ur  is  Store_name_value
bakery+cafe  is  Store_name_value
FlourBakery+Cate  is  Store_name_value
190.Massachusetts  is  Store_addr_value
Cambridge,_ma.02139  is  Store_addr_value
04/18/21/4:11  is  Date_value
$6.50  is  Prod_price_value
half-sandwich  is  Prod_item_value
46.50  is  Subtotal_value
Subtotal  is  Subtotal_key
$0.46  is  Tax_value
Jax  is  Tax_key
$6.96  is  Total_value
Amount  is  Subtotal_key
$0.00  is  Tips_value
Tip;  is  Tips_key
16.96  is  Total_value
Tatal  is  Total_key
```
I am very confused why my custom model's text recognition is generating worse results. For example, why is '46' being identified as the tax value, when the text identified by my model is '$0.46' (this means that the KIE model actually breaks up '46' from '$0.46', why is it doing that?). The case is the same for '96'  identified as Total_value (KIE is splitting it from '16.96' or ‘$6.96’)
Besides the price strings being incorrectly split by KIE, there is the issue of so many missing fields, such as the product name, and Store name (it only identified 'store' and not 'Flour'), address, date, time, etc.

Here is the list of boxes/text I'm passing to KIE:
`````
[{'box': [122, 1438, 323, 1438, 323, 1412, 122, 1412], 'box_score': 58, 'text': 'http://flourbakery', 'text_score': 0.9989816612667508}, {'box': [325, 1446, 456, 1446, 456, 1411, 325, 1411], 'box_score': 58, 'text': 'com/careers', 'text_score': 0.9666266603903337}, {'box': [115, 611, 163, 611, 163, 589, 115, 589], 'box_score': 95, 'text': 'with', 'text_score': 0.9999999701976776}, {'box': [173, 611, 230, 611, 230, 588, 173, 588], 'box_score': 95, 'text': 'Flour', 'text_score': 0.9995185971260071}, {'box': [239, 613, 307, 613, 307, 589, 239, 589], 'box_score': 95, 'text': 'Bakery', 'text_score': 0.9999999602635702}, {'box': [317, 610, 395, 610, 395, 589, 317, 589], 'box_score': 95, 'text': 'Rewards', 'text_score': 0.9999994039535522}, {'box': [404, 612, 446, 612, 446, 588, 404, 588], 'box_score': 95, 'text': 'App!', 'text_score': 0.9677503257989883}, {'box': [57, 1380, 112, 1380, 112, 1357, 57, 1357], 'box_score': 96, 'text': 'Thank', 'text_score': 1.0}, {'box': [122, 1383, 157, 1383, 157, 1363, 122, 1363], 'box_score': 96, 'text': 'you', 'text_score': 0.9999425411224365}, {'box': [166, 1381, 201, 1381, 201, 1357, 166, 1357], 'box_score': 96, 'text': 'for', 'text_score': 1.0}, {'box': [211, 1383, 301, 1383, 301, 1358, 211, 1358], 'box_score': 96, 'text': 'visiting', 'text_score': 0.9999790191650391}, {'box': [312, 1380, 371, 1380, 371, 1356, 312, 1356], 'box_score': 96, 'text': 'Flour', 'text_score': 0.9999343633651734}, {'box': [102, 589, 161, 589, 161, 562, 102, 562], 'box_score': 54, 'text': 'Order', 'text_score': 0.9995850563049317}, {'box': [171, 585, 230, 585, 230, 562, 171, 562], 'box_score': 76, 'text': 'ahead', 'text_score': 0.9999865293502808}, {'box': [240, 582, 251, 582, 251, 569, 240, 569], 'box_score': 96, 'text_score': 96}, {'box': [261, 585, 308, 585, 308, 567, 261, 567], 'box_score': 92, 'text': 'earn', 'text_score': 0.9999895989894867}, {'box': [317, 585, 372, 585, 372, 564, 317, 564], 'box_score': 95, 'text': 'sweet', 'text_score': 0.9909686207771301}, {'box': [385, 584, 461, 584, 461, 562, 385, 562], 'box_score': 96, 'text': 'rewards', 'text_score': 0.9999989782060895}, {'box': [102, 530, 140, 530, 140, 500, 102, 500], 'box_score': 96, 'text': 'Did', 'text_score': 0.9999999602635702}, {'box': [149, 532, 186, 532, 186, 512, 149, 512], 'box_score': 96, 'text': 'you', 'text_score': 0.9816183646519979}, {'box': [196, 530, 253, 530, 253, 508, 196, 508], 'box_score': 91, 'text': 'order', 'text_score': 0.9996009230613708}, {'box': [264, 530, 344, 530, 344, 507, 264, 507], 'box_score': 95, 'text': 'online?', 'text_score': 0.9999847752707345}, {'box': [354, 530, 411, 530, 411, 507, 354, 507], 'box_score': 96, 'text': 'Check', 'text_score': 1.0}, {'box': [154, 560, 245, 560, 245, 535, 154, 535], 'box_score': 88, 'text': 'register', 'text_score': 0.9999999850988388}, {'box': [254, 558, 301, 558, 301, 537, 254, 537], 'box_score': 95, 'text': 'when', 'text_score': 0.999992161989212}, {'box': [311, 560, 346, 560, 346, 540, 311, 540], 'box_score': 96, 'text': 'you', 'text_score': 0.9999783039093018}, {'box': [355, 558, 430, 558, 430, 537, 355, 537], 'box_score': 96, 'text': 'arrive!', 'text_score': 0.9999206151281085}, {'box': [165, 479, 199, 479, 199, 456, 165, 456], 'box_score': 96, 'text': '190', 'text_score': 0.9999492963155111}, {'box': [207, 479, 355, 479, 355, 456, 207, 456], 'box_score': 96, 'text': 'Massachusetts', 'text_score': 0.9999998991306012}, {'box': [364, 479, 398, 479, 398, 457, 364, 457], 'box_score': 96, 'text': 'Ave', 'text_score': 0.9999801317850748}, {'box': [100, 1465, 156, 1465, 156, 1440, 100, 1440], 'box_score': 96, 'text': 'guest', 'text_score': 0.9999943017959595}, {'box': [166, 1462, 209, 1462, 209, 1439, 166, 1439], 'box_score': 95, 'text': 'wifi', 'text_score': 0.9999955892562866}, {'box': [221, 1465, 320, 1465, 320, 1439, 221, 1439], 'box_score': 96, 'text': 'password:', 'text_score': 0.9937534199820625}, {'box': [354, 925, 368, 925, 368, 902, 354, 902], 'box_score': 92, 'text': 'C', 'text_score': 0.9479581713676453}, {'box': [379, 926, 422, 926, 422, 902, 379, 902], 'box_score': 85, 'text': '(EMV', 'text_score': 0.9999737739562988}, {'box': [431, 927, 476, 927, 476, 902, 431, 902], 'box_score': 96, 'text': 'Chip', 'text_score': 0.9999996423721313}, {'box': [485, 927, 544, 927, 544, 896, 485, 896], 'box_score': 86, 'text': 'Read)', 'text_score': 0.9999900579452514}, {'box': [360, 721, 447, 721, 447, 696, 360, 696], 'box_score': 89, 'text': '04/18/21', 'text_score': 0.9999800622463226}, {'box': [459, 719, 501, 719, 501, 697, 459, 697], 'box_score': 89, 'text': '4:11', 'text_score': 0.9988068342208862}, {'box': [513, 719, 537, 719, 537, 697, 513, 697], 'box_score': 95, 'text': 'PM', 'text_score': 0.9974032342433929}, {'box': [173, 507, 285, 507, 285, 483, 173, 483], 'box_score': 96, 'text': 'Cambridge,', 'text_score': 0.9999961972236633}, {'box': [297, 505, 321, 505, 321, 484, 297, 484], 'box_score': 96, 'text': 'MA', 'text_score': 0.9985694885253906}, {'box': [331, 505, 388, 505, 388, 478, 331, 478], 'box_score': 94, 'text': '02139', 'text_score': 0.9999899864196777}, {'box': [185, 1492, 273, 1492, 273, 1469, 185, 1469], 'box_score': 95, 'text': 'restroom', 'text_score': 0.9993861615657806}, {'box': [282, 1493, 337, 1493, 337, 1470, 282, 1470], 'box_score': 90, 'text': 'code:', 'text_score': 0.9987509727478028}, {'box': [349, 1492, 396, 1492, 396, 1469, 349, 1469], 'box_score': 90, 'text': '5858', 'text_score': 0.9956018775701523}, {'box': [174, 451, 232, 451, 232, 428, 174, 428], 'box_score': 49, 'text': 'Flour', 'text_score': 0.9999571204185486}, {'box': [242, 454, 310, 454, 310, 428, 242, 428], 'box_score': 96, 'text': 'Bakery', 'text_score': 1.0}, {'box': [320, 449, 332, 449, 332, 433, 320, 433], 'box_score': 96, 'text': '+', 'text_score': 0.9774360060691833}, {'box': [342, 451, 388, 451, 388, 428, 342, 428], 'box_score': 79, 'text': 'Cafe', 'text_score': 0.9211844503879547}, {'box': [238, 284, 270, 284, 270, 247, 238, 247], 'box_score': 70, 'text': 'L', 'text_score': 0.6556800007820129}, {'box': [182, 287, 244, 287, 244, 216, 182, 216], 'box_score': 39, 'text': '9', 'text_score': 0.9064330458641052}, {'box': [212, 366, 296, 366, 296, 246, 212, 246], 'box_score': 37, 'text': 'In', 'text_score': 0.41097915172576904}, {'box': [389, 1108, 542, 1108, 542, 1089, 389, 1089], 'box_score': 1, 'text': 'A000000031010', 'text_score': 0.9528939082072332}, {'box': [370, 1151, 394, 1151, 394, 1133, 370, 1133], 'box_score': 13, 'text': '8b', 'text_score': 0.9986675083637238}, {'box': [394, 1155, 548, 1155, 548, 1133, 394, 1133], 'box_score': 13, 'text': '16ed8646761e1b', 'text_score': 0.9357285627296993}, {'box': [32, 1013, 156, 1013, 156, 990, 32, 990], 'box_score': 92, 'text': 'Transaction', 'text_score': 0.9999977675351229}, {'box': [166, 1013, 211, 1013, 211, 984, 166, 984], 'box_score': 96, 'text': 'Type', 'text_score': 1.0}, {'box': [333, 1461, 389, 1461, 389, 1438, 333, 1438], 'box_score': 77, 'text': 'flour', 'text_score': 0.9999999046325684}, {'box': [391, 1461, 479, 1461, 479, 1437, 391, 1437], 'box_score': 75, 'text': 'lovesyou', 'text_score': 0.9984621182084084}, {'box': [410, 1091, 542, 1091, 542, 1064, 410, 1064], 'box_score': 43, 'text': 'xr7FfffyH.ighdm', 'text_score': 0.8578246335188547}, {'box': [303, 403, 384, 403, 384, 371, 303, 371], 'box_score': 96, 'text': 'bakery', 'text_score': 1.0}, {'box': [392, 397, 408, 397, 408, 380, 392, 380], 'box_score': 95, 'text': '+', 'text_score': 0.984312117099762}, {'box': [415, 397, 465, 397, 465, 371, 415, 371], 'box_score': 95, 'text': 'cafe', 'text_score': 0.9999856948852539}, {'box': [409, 951, 540, 951, 540, 929, 409, 929], 'box_score': 11, 'text': 'XXXXXXXX6600', 'text_score': 0.951850434144338}, {'box': [221, 1328, 359, 1328, 359, 1292, 221, 1292], 'box_score': -1, 'text': 'PARULASINGH', 'text_score': 0.9540803080255335}, {'box': [378, 1379, 447, 1379, 447, 1353, 378, 1353], 'box_score': 96, 'text': 'Bakery', 'text_score': 0.9999990065892538}, {'box': [457, 1374, 468, 1374, 468, 1358, 457, 1358], 'box_score': 93, 'text': '+', 'text_score': 0.49990764260292053}, {'box': [478, 1376, 524, 1376, 524, 1353, 478, 1353], 'box_score': 92, 'text': 'Cafe', 'text_score': 0.9999719262123108}, {'box': [162, 1408, 206, 1408, 206, 1384, 162, 1384], 'box_score': 95, 'text': 'life', 'text_score': 0.9999964535236359}, {'box': [216, 1408, 303, 1408, 303, 1384, 216, 1384], 'box_score': 96, 'text': 'sweeter,', 'text_score': 0.9976383298635483}, {'box': [35, 1067, 121, 1067, 121, 1038, 35, 1038], 'box_score': 96, 'text': 'Approval', 'text_score': 0.9973867163062096}, {'box': [133, 1067, 179, 1067, 179, 1045, 133, 1045], 'box_score': 96, 'text': 'Code', 'text_score': 0.9999583959579468}, {'box': [316, 1406, 362, 1406, 362, 1383, 316, 1383], 'box_score': 96, 'text': 'work', 'text_score': 0.9999996423721313}, {'box': [372, 1406, 394, 1406, 394, 1384, 372, 1384], 'box_score': 96, 'text': 'at', 'text_score': 0.9999986886978149}, {'box': [406, 1406, 470, 1406, 470, 1382, 406, 1382], 'box_score': 96, 'text': 'Flour!', 'text_score': 0.9999502599239349}, {'box': [33, 1041, 178, 1041, 178, 1017, 33, 1017], 'box_score': 76, 'text': 'Authorization', 'text_score': 0.9999930491814246}, {'box': [27, 962, 149, 962, 149, 929, 27, 929], 'box_score': -1, 'text': 'VISADEBIT', 'text_score': 0.9404886629846361}, {'box': [45, 1142, 165, 1142, 165, 1115, 45, 1115], 'box_score': 65, 'text': 'Application', 'text_score': 0.9999892928383567}, {'box': [173, 1136, 227, 1136, 227, 1118, 173, 1118], 'box_score': 95, 'text': 'Label', 'text_score': 0.998243761062622}, {'box': [41, 1119, 162, 1119, 162, 1092, 41, 1092], 'box_score': 95, 'text': 'Application', 'text_score': 0.9999957301399924}, {'box': [171, 1115, 194, 1115, 194, 1095, 171, 1095], 'box_score': 77, 'text': 'ID', 'text_score': 0.9799444377422333}, {'box': [19, 699, 147, 699, 147, 667, 19, 667], 'box_score': -1, 'text': 'Checkmas30', 'text_score': 0.8507950335741044}, {'box': [24, 775, 76, 775, 76, 752, 24, 752], 'box_score': 50, 'text': 'half', 'text_score': 0.9997201412916183}, {'box': [87, 775, 177, 775, 177, 753, 87, 753], 'box_score': 93, 'text': 'sandwich', 'text_score': 0.9999192580580711}, {'box': [49, 1182, 95, 1182, 95, 1159, 49, 1159], 'box_score': 93, 'text': 'Card', 'text_score': 0.9999982714653015}, {'box': [101, 1182, 169, 1182, 169, 1152, 101, 1152], 'box_score': 96, 'text': 'Reader', 'text_score': 0.999998946984609}, {'box': [26, 671, 106, 671, 106, 644, 26, 644], 'box_score': 95, 'text': 'Server:', 'text_score': 0.9999995912824359}, {'box': [118, 667, 178, 667, 178, 644, 118, 644], 'box_score': 96, 'text': 'staff', 'text_score': 0.9999999046325684}, {'box': [433, 1129, 478, 1129, 478, 1111, 433, 1111], 'box_score': 86, 'text': 'VISA', 'text_score': 0.9957083761692047}, {'box': [487, 1129, 542, 1129, 542, 1111, 487, 1111], 'box_score': 95, 'text': 'DEBIT', 'text_score': 0.9997801065444947}, {'box': [452, 1036, 542, 1036, 542, 1012, 452, 1012], 'box_score': 96, 'text': 'Approved', 'text_score': 0.9999984353780746}, {'box': [28, 724, 117, 724, 117, 700, 28, 700], 'box_score': 96, 'text': 'Ordered:', 'text_score': 0.9617960453033447}, {'box': [476, 1059, 541, 1059, 541, 1037, 476, 1037], 'box_score': 86, 'text': '151216', 'text_score': 0.9996478656927744}, {'box': [148, 372, 202, 372, 202, 265, 148, 265], 'box_score': -1, 'text': 'l', 'text_score': 0.9382102489471436}, {'box': [481, 1294, 551, 1294, 551, 1258, 481, 1258], 'box_score': -1, 'text': '16.96', 'text_score': 0.992550003528595}, {'box': [489, 1261, 545, 1261, 545, 1235, 489, 1235], 'box_score': 71, 'text': '$0.00', 'text_score': 0.9992386937141419}, {'box': [36, 1094, 114, 1094, 114, 1068, 36, 1068], 'box_score': 95, 'text': 'Payment', 'text_score': 0.9999928644725254}, {'box': [124, 1088, 147, 1088, 147, 1068, 124, 1068], 'box_score': 91, 'text': 'ID', 'text_score': 0.9509776830673218}, {'box': [481, 1236, 550, 1236, 550, 1198, 481, 1198], 'box_score': -1, 'text': '16.96', 'text_score': 0.9790594100952148}, {'box': [152, 1300, 231, 1300, 231, 1266, 152, 1266], 'box_score': -1, 'text': 'Total:', 'text_score': 0.9925598502159119}, {'box': [483, 772, 539, 772, 539, 750, 483, 750], 'box_score': 84, 'text': '$6.50', 'text_score': 0.9943578004837036}, {'box': [485, 870, 540, 870, 540, 848, 485, 848], 'box_score': 89, 'text': '$6.96', 'text_score': 0.9995059967041016}, {'box': [366, 365, 424, 365, 424, 288, 366, 288], 'box_score': -1, 'text': 'r', 'text_score': 0.9985647797584534}, {'box': [489, 1176, 544, 1176, 544, 1153, 489, 1153], 'box_score': 35, 'text': 'BBPOS', 'text_score': 0.987549340724945}, {'box': [35, 825, 125, 825, 125, 801, 35, 801], 'box_score': 65, 'text': 'Subtotal', 'text_score': 0.9999233335256577}, {'box': [143, 1242, 223, 1242, 223, 1210, 143, 1210], 'box_score': -1, 'text': 'Amount', 'text_score': 0.9999992847442627}, {'box': [485, 821, 540, 821, 540, 800, 485, 800], 'box_score': 88, 'text': '$6.50', 'text_score': 0.9952187418937684}, {'box': [480, 690, 534, 690, 534, 667, 480, 667], 'box_score': 88, 'text': 'Parul', 'text_score': 0.9995231986045837}, {'box': [485, 845, 540, 845, 540, 824, 485, 824], 'box_score': 31, 'text': '$0.46', 'text_score': 0.9635659456253052}, {'box': [506, 1004, 541, 1004, 541, 981, 506, 981], 'box_score': 74, 'text': 'ale', 'text_score': 0.9999911387761434}, {'box': [43, 875, 96, 875, 96, 853, 43, 853], 'box_score': 93, 'text': 'Total', 'text_score': 0.9998692870140076}, {'box': [105, 1408, 151, 1408, 151, 1385, 105, 1385], 'box_score': 89, 'text': 'Make', 'text_score': 0.9999999701976776}, {'box': [48, 1160, 134, 1160, 134, 1137, 48, 1137], 'box_score': 37, 'text': 'Terminal', 'text_score': 0.9999392181634903}, {'box': [175, 1272, 231, 1272, 231, 1235, 175, 1235], 'box_score': -1, 'text': 'Tip:', 'text_score': 0.9353603720664978}, {'box': [28, 936, 97, 936, 97, 900, 28, 900], 'box_score': -1, 'text': 'Input', 'text_score': 0.9999919652938842}, {'box': [89, 370, 127, 370, 127, 303, 89, 303], 'box_score': -1, 'text': 'want', 'text_score': 0.760039821267128}, {'box': [102, 930, 147, 930, 147, 906, 102, 906], 'box_score': 96, 'text': 'Type', 'text_score': 0.9999983012676239}, {'box': [40, 848, 77, 848, 77, 824, 40, 824], 'box_score': 68, 'text': 'Jax', 'text_score': 0.8586016297340393}, {'box': [92, 314, 127, 314, 127, 271, 92, 271], 'box_score': -1, 'text': 'the', 'text_score': 0.5911299536625544}, {'box': [111, 561, 146, 561, 146, 530, 111, 530], 'box_score': -1, 'text': 'at', 'text_score': 0.9999984502792358}, {'box': [-10, 1531, 46, 1531, 46, 1444, -10, 1444], 'box_score': -1, 'text': 'the', 'text_score': 0.5540283421675364}, {'box': [416, 534, 448, 534, 448, 502, 416, 502], 'box_score': -1, 'text': 'in', 'text_score': 0.9999934434890747}, {'box': [141, 1157, 169, 1157, 169, 1137, 141, 1137], 'box_score': 69, 'text': 'ID', 'text_score': 0.9744626879692078}]
`````

**Comment by gaotongxiao** [comment]
Did you use the same pipeline provided in `ocr.py` when you passed your list to the KIE model? The text processing pipeline should not split up `$0.46` with the same dictionary. 
For your second question, the KIE model was pretrained on the annotations where the text boxes are segmented on the line level instead of the word level. Therefore it performs better on `PS_CTW`, which tends to generate line-level text boxes. You can try merging the boxes from your model using `stitch_boxes_into_lines` (https://mmocr.readthedocs.io/en/latest/api.html#mmocr.utils.stitch_boxes_into_lines). The function, however, is only experimental, and any suggestions are appreicated.

**Comment by parulsingh23** [comment]
Thank you so much for the suggestions!
For the first point, since I don't want to use the end-to-end pipeline and am just interested in KIE, I created a new function in ocr.py (similar to `det_recog_kie_inference`). Essentially, I pass in an args called 'result' which is the list of dictionaries for the identified text in thee image to readText. This new function then gets called when only the KIE model is asked for (so no text det/recog model instantiated in the MMOCR object).
Since this is the case, I looked in my passed in list of dictionaries (the 'result' arg) and I found only 1 text that included '46', and that was:
`{'box': [485, 845, 540, 845, 540, 824, 485, 824], 'box_score': 31, 'text': '$0.46', 'text_score': 0.9635659456253052}`
As you can see, if '46' was provided by the KIE model, and _no other_ dictionary had a text containing '46', then the KIE model _must_ have split it off from this text. I don't understand why this behavior would occur.

**Comment by gaotongxiao** [comment]
Could you share the original test image so that I can test it through? I'm also not sure why the splitting happens.

For your second question, SDMGR has been trained on the annotations without any spaces. That is, those neighboring words on your receipt should also be joined _without spaces_. For example, "half" and "sandwich" should be joined as "halfsandwich". You may take a look at WildReceipt for examples.

> I do want to clarify, however, what is considered inline. For example, in a receipt i can have the following line:
> `Boy Shoes\t\t$18.99`
> which I think you were saying (for the purposes of the KIE) it should be broken down and delivered as 'BoyShoes' and '$18.99'. (Please correct me if I'm wrong)

You're right.

> I am concerned, however, about the scenario where a product quantity is specified:
> `2 Boy Shoes\t\t$37.98`
> In this case, if I send '2BoyShoes' and '$37.98', would the KIE be able to 'split' the 2 off from the product name, and identify it separately as the product quantity?

I think it's totally fine, SDMGR is likely to identify "2BoyShoes" as the product name.

**Comment by parulsingh23** [comment]
Thank you for the follow-up! Here's the original image: 
![IMG_20210425_181953014](https://user-images.githubusercontent.com/55006754/128817388-86be5390-22bf-4181-a20a-a409b57915cf.jpg)

**Comment by gaotongxiao** [comment]
I think you have modified the code somewhere else - I used your picture and your annotations in ocr.py. For the quick check, I had inserted the following line to ocr.py, line 553:
```
img_e2e_res['result'] = your annotations
```
Then `kie_dataset._parse_anno_info` raised an error complaining about missing the `text` entry in one element. I doubt that you were using something else to parse the annotation info? As far as I know, `_parse_anno_info` doesn't split texts.
